lib/tapioca/generators/gem.rb

```diff
@@ -0,0 +1,372 @@
+        say_error("\n\nLoadError: #{error}", :bold, :red)
+        say_error("\nTapioca could not load all the gems required by your application.", :yellow)
+        say_error("If you populated ", :yellow)
+        say_error("#{file} ", :bold, :blue)
+        say_error("with ", :yellow)
+        say_error("`#{Config::DEFAULT_COMMAND} require`", :bold, :blue)
+        say_error("you should probably review it and remove the faulty line.", :yellow)
+      end
+
+      EMPTY_RBI_COMMENT = <<~CONTENT
+        # THIS IS AN EMPTY RBI FILE.
+        # see https://github.com/Shopify/tapioca/wiki/Manual-Gem-Requires
+      CONTENT
+
+      sig { params(command: String, reason: T.nilable(String), strictness: T.nilable(String)).returns(String) }
+      def rbi_header(command, reason: nil, strictness: nil)
+        statement = <<~HEAD
+          # DO NOT EDIT MANUALLY
+          # This is an autogenerated file for #{reason}.
+          # Please instead update this file by running `#{command}`.
+        HEAD
+
+        sigil = <<~SIGIL if strictness
+          # typed: #{strictness}
+        SIGIL
+
+        if config.file_header
+          [statement, sigil].compact.join("\n").strip.concat("\n\n")
+        elsif sigil
+          sigil.strip.concat("\n\n")
+        else
+          ""
+        end
+      end
+
+      sig { params(filename: Pathname).void }
+      def remove(filename)
+        return unless filename.exist?
+        say("-- Removing: #{filename}")
+        filename.unlink
+      end
+
+      sig { void }
+      def perform_sync_verification
+        diff = {}
+
+        removed_rbis.each do |gem_name|
+          filename = existing_rbi(gem_name)
+          diff[filename] = :removed
+        end
+
+        added_rbis.each do |gem_name|
+          filename = expected_rbi(gem_name)
+          diff[filename] = gem_rbi_exists?(gem_name) ? :changed : :added
+        end
+
+        report_diff_and_exit_if_out_of_date(diff, "gem")
+      end
+
+      sig { params(diff: T::Hash[String, Symbol], command: String).void }
+      def report_diff_and_exit_if_out_of_date(diff, command)
+        if diff.empty?
+          say("Nothing to do, all RBIs are up-to-date.")
+        else
+          say("RBI files are out-of-date. In your development environment, please run:", :green)
+          say("  `#{Config::DEFAULT_COMMAND} #{command}`", [:green, :bold])
+          say("Once it is complete, be sure to commit and push any changes", :green)
+
+          say("")
+
+          say("Reason:", [:red])
+          diff.group_by(&:last).sort.each do |cause, diff_for_cause|
+            say(build_error_for_files(cause, diff_for_cause.map(&:first)))
+          end
+
+          exit(1)
+        end
+      end
+
+      sig { void }
+      def perform_removals
+        say("Removing RBI files of gems that have been removed:", [:blue, :bold])
+        puts
+
+        anything_done = T.let(false, T::Boolean)
+
+        gems = removed_rbis
+
+        shell.indent do
```

Comment:
The `indent` method was previously accessible due to our subclassing of `Thor::Shell`. So now I've patched it to call the instance of `shell` that lives on `Thor::Base` in `Generators::Base`.